.github/workflows/main.yml

```diff
@@ -2,66 +2,193 @@ name: main
 
 on:
     push:
-        branches:
-            - main
+        branches: [main]
     pull_request:
 
 concurrency:
     group: ${{ github.workflow }}-${{ github.head_ref || github.run_id }}
     cancel-in-progress: true
 
 jobs:
-    style:
+    deps:
```

Comment:
How does this job work exactly?

Looks like the workflow doesn't install dependencies until there's some change in the yarn.lock file right?

So, I imagine the other jobs like tests, style and compile should be quite faster right?

Reply:
My understanding is that job contexts are isolated.
So when you do e.g
```yaml
job:
  steps:
    - uses: actions/setup-node@v4
      with:
        cache: yarn
```

It only caches deps in this particular job.
As the workflow is running several jobs  that depends on the same lock file (we are in a monorepo), I thought it is redundant to run `yarn` in all jobs and was looking for a way to share the `node_modules` content **between jobs**.
One solution is to use directly `actions/cache` (`actions/setup-node` cache feature uses it under the hood).
It should speed up the whole workflow but I haven't measured it.

Reply:
I see! Looks great and we should definitely re-use it in the other repos imo.